packages/api/src/resolvers/type/media-types.ts

```diff
@@ -1,12 +1,26 @@
-import { File as FileType } from '@unchainedshop/core-files';
+import { File as FileType, getFileAdapter } from '@unchainedshop/core-files';
 import { Context } from '../../context.js';
-
+import { checkAction } from '../../acl.js';
+import { actions } from '../../roles/index.js';
 export interface MediaHelperTypes {
-  url: (language: FileType, params: Record<string, any>, context: Context) => string;
+  url: (language: FileType, params: Record<string, any>, context: Context) => Promise<string>;
 }
 
 export const Media: MediaHelperTypes = {
-  url(root, params, { modules }) {
-    return modules.files.getUrl(root, params);
+  url: async (file, params, context) => {
+    const { modules } = context;
+    try {
+      await checkAction(context, actions.downloadFile, [file, params]);
+      const fileUploadAdapter = getFileAdapter();
+      const mediaUrl = modules.files.getUrl(file, params);
+      if (file.isPrivate) {
+        return fileUploadAdapter.signUrl(mediaUrl, file._id);
```

Comment:
i think we should get rid of the isPrivate and let the provider decide the way this is handled based on a meta property.

So this code should be more or less:

const url = Adapter.buildDownloadUrl()
return fileSettings.transformUrl(url)

please move the getUrl handling to the setting default fn.